vmdb/app/models/mixins/container_provider_mixin.rb

```diff
@@ -0,0 +1,42 @@
+module ContainerProviderMixin
+  extend ActiveSupport::Concern
+
+  included do
+
+    has_many :container_nodes,                      :foreign_key => :ems_id, :dependent => :destroy
+    has_many :container_groups,                     :foreign_key => :ems_id, :dependent => :destroy
+    has_many :container_services,                   :foreign_key => :ems_id, :dependent => :destroy
+
+    def self.raw_api_endpoint(hostname, port)
+      URI::HTTPS.build(:host => hostname, :port => port.to_i)
+    end
+
+    # UI methods for determining availability of fields
+    def supports_port?
```

Comment:
These instance methods should live outside the `included` and just be normal module instance methods at the `ContainerProviderMixin` level